docs/adr/0016-auth-for-settings-redesign.md

```diff
@@ -0,0 +1,114 @@
+
+## Decision Drivers
+
+- Smooth UX
+- Security
+- Development velocity
+- Ease of integration
+
+## Considered Options
+
+- Authenticate with sessionToken, then use a unique Bearer token for auth-server APIs.
+- Authenticate via OAuth2 to use auth-server APIs
+- Authenticate via OAuth2 to use DB directly
+- Reuse existing sessionToken on content-server with auth-server APIs
+- Implement own login sessions
+
+## Decision Outcome
+
+Chosen option: "Authenticate with sessionToken, then use a unique Bearer token for auth-server APIs", because:
+
+- It requires minimal changes to existing auth-server implementation and infrastructure.
+- It relies on an authentication mechanism with relatively well-known security properties.
+
+### Positive Consequences
+
+- We do not have to re-implement substantial logic in the auth-server for using OAuth2 tokens.
+
+### Negative Consequences
+
+- We will need to modify auth strategies in auth-server APIs.
+- We don't have an entirely greenfield project, which could be a bit of a drag. But, rediscovering lessons learned in existing code can also be a drag.
+- We will be using a unique Bearer token between the graphql API and the auth-server that purports to represent the original Hawk request.
+
+## Pros and Cons of the Options
+
+### Authenticate with sessionToken, then use a unique Bearer token for auth-server APIs.
+
+While we could like to someday deprecate the session token mechanism, in this case a substantial portion of the API's used in auth-server assume a sessionToken and the data returned when authenticating it will be used. The session token includes additional relevant authentication method information that is currently not present on the OAuth2 tokens which further complicates adapting the auth-server methods to use an OAuth 2 token.
+
+- Pros:
+  - Can be done with seamless UX - user visits new settings without confirmation
+  - Minimal work involved as the new auth-server auth strategy can lookup originating sessionTokenID.
+- Neutral:
+  - GraphQL server needs to proxy auth-server APIs
+- Cons:
+  - More systems that continue to use sessionTokens which we would like to deprecate.
+  - We use Hawk request signing with the session token credentials for authentication. Passing these credentials from the frontend to a GQL backend essentially defeats the purpose of Hawk, since the scheme is in part meant to minimize the transmission of credentials over the network.
```

Comment:
This seems like a copy from "Reuse existing sessionToken". Should be different here, if I understand the new proposed mechanism:

We wouldn't send the Hawk sessionToken credentials over the network. Instead, we'd generate and send a unique Bearer token that's one step removed from those original credentials. 

That should mean we're using a different mechanism, not defeating Hawk in general. Maybe a subtle difference, but worth highlighting?